subiquity/models/filesystem.py

```diff
@@ -218,24 +218,38 @@ def dehumanize_size(size):
     return num * mult // div
 
 
-# This is a guess!
-RAID_OVERHEAD = 8 * (1 << 20)
-
-
 def get_raid_size(level, devices):
     if len(devices) == 0:
         return 0
-    min_size = min(dev.size for dev in devices) - RAID_OVERHEAD
+    # The calculation of how much of a device mdadm uses for raid is a
+    # touch ridiculous. What follows is a translation of the code at:
+    # https://git.kernel.org/pub/scm/utils/mdadm/mdadm.git/tree/super1.c?h=mdadm-4.1&id=20e8fe52e7190b3ffda127566852eac2eb7fa1f7#n2770
+    # (note that that calculation is in terms of 512-byte sectors and
+    # this one is in bytes).
+    #
+    # This makes assumptions about the defaults mdadm uses but mostly
+    # that the default metadata version is 1.2, and other formats use
+    # less space.
+    min_size = min(dev.size for dev in devices)
+    bmspace = 128*1024
```

Comment:
I'd expand the variable names here, since it's not kernel code.
bitmap_space, and summarize the decision, we're assuming
most devices are going to be > 200GiB so we select the 128k bitmap_space

Reply:
I think there's some value in following the variable names from the mdadm source (maybe I should paste that in a comment) but you're right about where the value for bmspace comes from.